packages/ddp/livedata_connection.js

```diff
@@ -520,12 +532,41 @@ _.extend(Connection.prototype, {
         if (!existing.ready)
           existing.readyCallback = callbacks.onReady;
       }
+
+      // XXX COMPAT WITH 1.0.3.1 we used to have onError but now we call
+      // onStopped with an optional error argument
       if (callbacks.onError) {
         // Replace existing callback if any, so that errors aren't
         // double-reported.
         existing.errorCallback = callbacks.onError;
       }
+
+      if (callbacks.onStopped) {
+        existing.stoppedCallback = callbacks.onStopped;
+      }
+
+      // Return the exact same handle so that we can use === to check if the
```

Comment:
well, i mean, just use a map instead of an array?  then it actually becomes easier?  (or even ReactiveDict instead of ReactiveVar if that makes sense)